common/djangoapps/embargo/tests/test_api.py

```diff
@@ -0,0 +1,147 @@
+        self.embargo_course1.save()
+        self.embargo_course2 = CourseFactory.create()
+        self.embargo_course2.save()
+        self.regular_course = CourseFactory.create(org="Regular")
+        self.regular_course.save()
+        self.embargoed_course_whitelisted = '/courses/' + self.embargo_course1.id.to_deprecated_string() + '/info'
+        self.embargoed_course_blacklisted = '/courses/' + self.embargo_course2.id.to_deprecated_string() + '/info'
+        self.regular_page = '/courses/' + self.regular_course.id.to_deprecated_string() + '/info'
+
+        restricted_course_1 = RestrictedCourse(course_key=self.embargo_course1.id)
+        restricted_course_1.save()
+
+        restricted_course_2 = RestrictedCourse(course_key=self.embargo_course2.id)
+        restricted_course_2.save()
+
+        all_countries = [Country(country=code[0]) for code in list(countries)]
+        Country.objects.bulk_create(all_countries)
+
+        country_access_white_rules = [
+            CountryAccessRule(
+                restricted_course=restricted_course_1,
+                rule_type=WHITE_LIST,
+                country=Country.objects.get(country='US')
+            ),
+            CountryAccessRule(
+                restricted_course=restricted_course_1,
+                rule_type=WHITE_LIST,
+                country=Country.objects.get(country='NZ')
+            )
+        ]
+        CountryAccessRule.objects.bulk_create(country_access_white_rules)
+
+        country_access_black_rules = [
+            CountryAccessRule(
+                restricted_course=restricted_course_2,
+                rule_type=BLACK_LIST,
+                country=Country.objects.get(country='IR')
+            ),
+            CountryAccessRule(
+                restricted_course=restricted_course_2,
+                rule_type=BLACK_LIST,
+                country=Country.objects.get(country='AF')
+            )
+        ]
+        CountryAccessRule.objects.bulk_create(country_access_black_rules)
+
+        # Text from lms/templates/static_templates/embargo.html
+        self.embargo_text = "Unfortunately, at this time edX must comply with export controls, and we cannot allow you to access this course."
+
+        self.patcher = mock.patch.object(pygeoip.GeoIP, 'country_code_by_addr', self.mock_country_code_by_addr)
+        self.patcher.start()
+
+    def tearDown(self):
+        # Explicitly clear ConfigurationModel's cache so tests have a clear cache
+        # and don't interfere with each other
+        cache.clear()
+        self.patcher.stop()
+
+    def mock_country_code_by_addr(self, ip_addr):
+        """
+        Gives us a fake set of IPs
+        """
+        ip_dict = {
+            '1.0.0.0': 'CU',
+            '2.0.0.0': 'IR',
+            '3.0.0.0': 'SY',
+            '4.0.0.0': 'SD',
+            '5.0.0.0': 'AQ',  # Antartica
+            '2001:250::': 'CN',
+            '2001:1340::': 'CU',
+        }
+        return ip_dict.get(ip_addr, 'US')
```

Comment:
There are a few more test cases to add here in order to get 100% coverage of the middleware function.

In addition, please use `self.assertNumQueries()` (see the [docs](https://docs.djangoproject.com/en/1.4/topics/testing/#django.test.TestCase.assertNumQueries)) to verify that caching is working correctly.  It should be possible to check that there are queries on the first request, but no queries on the second request (because it's using the cache).  (You can add these assertions to the test cases you've already written; you don't need to create new test cases specifically to test caching.)